### PR TITLE
3394 ordering not taking alias into account

### LINKED ISSUE #3394: Ordering not taking alias into account
**Describe the bug**
Sorting by `@alias` properties do not take into account aliased name, and instead sorts by property name.

**Type definitions**
```graphql
type Product {
  id: String! @alias(property: "fg_item_id")
  description: String!
  partNumber: ID! @alias(property: "fg_item")
}
```

**To Reproduce**
```
query listProducts {
  products(options: { sort: { partNumber: DESC } }) {
    id
    partNumber
    description
  }
}
```

Generates the Cypher:

```cypher
MATCH (this:`Product`)
WITH *
ORDER BY this.partNumber DESC
RETURN this { id: this.fg_item_id, partNumber: this.fg_item, .description } AS this
```

**Expected behavior**
The ordering should be using the aliased value:

```
MATCH (this:`Product`)
WITH *
ORDER BY this.fg_item DESC
RETURN this { id: this.fg_item_id, partNumber: this.fg_item, .description } AS this
```

**Additional context**
Original message: https://discord.com/channels/787399249741479977/1112848457015754904

### PR DESCRIPTION
**Pull request by angrykoala**
# Description
When sorting, we are using the options argument directly. This PR transforms the given fields to the DbFieldNames when needed to properly take `@alias` into account.

This PR adds a method `getFields` in Node which is not efficient and should be refactored into the SchemaModel

## Complexity

Complexity: Low

# Issue

Closes #3394

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
Made some changes to the Neo4j GraphQL Toolbox? Or made some changes to the core library perhaps?

Check out your changes at https://neo4j-graphql-toolbox-3396.surge.sh

Use for instance a local dbms hosted on [Neo4j Desktop](https://neo4j.com/download/) or a cloud-hosted [Aura Free](https://neo4j.com/cloud/aura-free/) dbms.

**Review by darrellwarde — CHANGES_REQUESTED** [review]
Needs a changeset, but other than that, looks good! The comment is nitpicky, you can ignore if you wish. 🙃

**Comment by neo4j-team-graphql** [comment]
# Performance Report

No Performance Changes


<details>
    <summary>Show Full Table</summary>
    
| name | dbHits | old dbHits | time (ms) | old time (ms) | maxRows |
| ---- | ------ | ---------- | --------- | ------------- | ------- |
|  aggregations.TopLevelAggregate | 3403 | 3403 | 42 | 48 | 1134 |
|  aggregations.NestedAggregation | 16553 | 16553 | 76 | 116 | 2174 |
|  aggregations.AggregationWithWhere | 11979 | 11979 | 52 | 55 | 2174 |
|  aggregations.AggregationWhereWithinNestedRelationships | 22116987 | 22116987 | 3362 | 3346 | 2008534 |
|  aggregations.AggregationWhereWithinNestedConnections | 22116987 | 22116987 | 2851 | 3033 | 2008534 |
|  aggregations.NestedCountFromMovieToActors | 9734 | 9734 | 42 | 58 | 2174 |
|  aggregations.NestedCountFromActorsToMovie | 9937 | 9937 | 42 | 57 | 2174 |
|  aggregations.DeeplyNestedCount | 13070331 | 13070333 | 4719 | 4787 | 2008534 |
|  batch-create.BatchCreate | 4200 | 4200 | 141 | 188 | 600 |
|  batch-create.BatchCreateSmall | 77 | 77 | 113 | 68 | 11 |
|  connect.createAndConnect | 14424 | 14424 | 228 | 258 | 3003 |
|  connections.Connection | 14082 | 14082 | 70 | 88 | 2174 |
|  connections.NestedConnection | 45466 | 45472 | 113 | 189 | 4516 |
|  connections.ConnectionWithSort | 3284 | 3284 | 65 | 103 | 1040 |
|  connections.ConnectionWithSortAndCypher | 3284 | 3284 | 193 | 259 | 1040 |
|  create.SimpleMutation | 7 | 7 | 68 | 62 | 1 |
|  cypher-directive.TopLevelCypherDirective | 122100 | 122100 | 747 | 794 | 12241 |
|  cypher-directive.TopLevelCypherDirectiveWithColumnName | 153619 | 153617 | 207 | 204 | 12241 |
|  delete.SimpleDelete | 19401 | 19401 | 296 | 271 | 1040 |
|  delete.NestedDeleteInUpdate | 23202 | 23202 | 150 | 214 | 2040 |
|  2925.SingleRelationshipFilter | 6468 | 6468 | 50 | 84 | 1040 |
|  2925.NestedSingleRelationshipFilter | 22900 | 22900 | 79 | 102 | 2174 |
|  2925.SingleRelationshipRequiredFilter | 5201 | 5201 | 40 | 42 | 1040 |
|  2925.NestedSingleRelationshipRequiredFilter | 9361 | 9361 | 57 | 85 | 1040 |
|  query.SimpleQuery | 3121 | 3121 | 24 | 31 | 1040 |
|  query.SimpleQueryWithRelationship | 16162 | 16162 | 45 | 52 | 2174 |
|  query.QueryWhere | 9704 | 9708 | 40 | 60 | 2163 |
|  query.SimpleQueryWithNestedWhere | 9882 | 9886 | 52 | 69 | 2163 |
|  query.Nested | 10096041 | 10096041 | 11343 | 11224 | 2008534 |
|  query.NestedWithFilter | 10074401 | 10074401 | 10991 | 11326 | 2004000 |
|  query.OrFilterOnRelationships | 42486 | 43544 | 190 | 276 | 1635 |
|  query.OrFilterOnRelationshipsAndNested | 36132 | 37303 | 238 | 305 | 1635 |
|  query.QueryWithNestedIn | 14041 | 14744 | 72 | 69 | 1430 |
|  query.NestedConnectionWhere | 9834 | 9834 | 55 | 70 | 2174 |
|  query.DeeplyNestedConnectionWhere | 10049 | 10049 | 91 | 91 | 2174 |
|  query.DeeplyNestedWithRelationshipFilters | 19344 | 19308 | 135 | 154 | 1664 |
|  query.NestedWithRelationshipSingleFilters | 3881 | 3881 | 206 | 358 | 1134 |
|  query.Fulltext | 80 | 80 | 31 | 47 | 16 |
|  query.FulltextWithNestedQuery | 587 | 587 | 49 | 72 | 84 |
|  sorting.SortMultipleTypes | 3515 | 3515 | 64 | 83 | 1040 |
|  sorting.SortMultipleTypesWithCypherWithCypher | 1468 | 1459 | 193 | 245 | 1040 |
|  sorting.SortOnNestedFields | 14082 | 14082 | 50 | 72 | 2174 |
|  sorting.SortDeeplyNestedFields | 43198 | 43198 | 91 | 115 | 4516 |
|  sorting.SortWithTopLevelCypher | 3121 | 3121 | 110 | 180 | 1040 |
|  unions.SimpleUnionQuery | 321 | 321 | 79 | 77 | 35 |
|  unions.SimpleUnionQueryWithMissingFields | 293 | 293 | 66 | 64 | 35 |
|  unions.NestedUnion | 410637 | 410637 | 335 | 316 | 33033 |
|  unions.NestedUnionWithMissingFields | 384611 | 384611 | 421 | 306 | 33033 |
|  update.NestedUpdate | 16143 | 16143 | 129 | 126 | 2002 |

</details>
        
Old Schema Generation: 40.637s
Schema Generation: 40.593s
Old Subgraph Schema Generation: 54.648s
Subgraph Schema Generation: 53.959s

File: packages/graphql/src/classes/GraphElement.ts [hunk 1]
```diff
@@ -59,4 +60,28 @@ export abstract class GraphElement {
...
+            this.pointFields,
+            this.customResolverFields,
+        ];
+    }
+
+    private searchField(fields: BaseField[], fieldName: string): BaseField | undefined {
+        return fields.find((field) => {
+            return field.fieldName === fieldName;
+        });
+    }
```
[the remark this answers is not in the document]
Comment (angrykoala) [reply packages/graphql/src/classes/GraphElement.ts:86]: The function `getField` without this extra function would look like this: 

```ts
    public getField(name: string): BaseField | undefined {
        for (const fieldList of this.getAllFields()) {
            const field = fields.find((field) => {
                return field.fieldName === fieldName;
            });
            if (field) return field;
        }
    }
```

A loop with a nested callback (which in essence is another loop) is a bit hard to read IMO, keep in mind that the helper is technically a single function but in truth is a loop basically.

 On second look, I would say that the naming was misleading though, the helper method should be called `searchFieldInList` to make it distinct from `getFields`

[the remark this answers is not in the document]
Comment (darrellwarde) [reply packages/graphql/src/classes/GraphElement.ts]: As I said, nitpicky, it's just my opinion, I'm happy whichever way.

**Review by darrellwarde — APPROVED** [review]

**Pull request merged**